ui/app/AppLayouts/Chat/ContactsColumn.qml

```diff
@@ -10,23 +10,233 @@ import "./components"
+        anchors.bottom: parent.bottom
         anchors.horizontalCenter: parent.horizontalCenter
-        font.weight: Font.Bold
-        font.pixelSize: 17
+
+        width: parent.width
+
+        leftPadding: Style.current.halfPadding
+        rightPadding: Style.current.halfPadding
+
+        ScrollBar.horizontal.policy: ScrollBar.AlwaysOff
+
+        contentHeight: channelList.height + 2 * Style.current.padding + emptyViewAndSuggestions.height + emptyViewAndSuggestions.anchors.topMargin
+        clip: true
+
+        Item {
+            id: noSearchResults
+            anchors.top: parent.top
+            anchors.horizontalCenter: parent.horizontalCenter
+            width: parent.width
+            visible: !!!channelList.height && contactsColumn.searchStr !== ""
+            height: visible ? 300 : 0
+
+            StatusBaseText {
+                font.pixelSize: 15
+                color: Theme.palette.directColor5
+                anchors.centerIn: parent
+                text: qsTr("No search results")
+            }
+        }
+
+        StatusChatList {
+            id: channelList
+
+            chatNameFn: function (chatItem) {
+                return chatItem.chatType !== Constants.chatTypePublic ?
+                    Emoji.parse(Utils.removeStatusEns(Utils.filterXSS(chatItem.name))) :
+                    Utils.filterXSS(chatItem.name)
+            }
+
+            profileImageFn: function (id) {
+                return appMain.getProfileImage(id)
+            }
+
+            filterFn: function (chatListItem) {
+                return !!!contactsColumn.searchStr || chatListItem.name.toLowerCase().includes(contactsColumn.searchStr.toLowerCase())
+            }
+
+            Connections {
+                target: profileModel.contacts.list
+                onContactChanged: {
+                    for (var i = 0; i < channelList.chatListItems.count; i++) {
+                        let chatItem = channelList.chatListItems.itemAt(i);
+                        if (chatItem.chatId === pubkey) {
+                            let profileImage = appMain.getProfileImage(pubkey)
+                            if (!!profileImage) {
+                                chatItem.image.isIdenticon = false
+                                chatItem.image.source = profileImage
+                            }
+                            break;
+                        }
+                    }
+                }
+            }
+
+            chatListItems.model: chatsModel.channelView.chats
+            selectedChatId: chatsModel.channelView.activeChannel.id
+
+            onChatItemSelected: chatsModel.channelView.setActiveChannel(id)
+            onChatItemUnmuted: chatsModel.channelView.unmuteChatItem(id)
+
+            popupMenu: StatusPopupMenu {
+
+                id: chatListContextMenu
+
+                property var chatItem
+
+                openHandler: function (id) {
+                    chatItem = chatsModel.channelView.getChatItemById(id)
```

Comment:
This is where we make use of that newly introduced API here: https://github.com/status-im/status-desktop/pull/2791/files#diff-9a0017eebdbf2f461ab80d57d59593fd3e335bc510fcc27e4ec3a7d7d77025eeR298

Previously, the channel context menu expected some object on `chatsModel`, but since `StatusPopupMenu` has no insight over `chatsModel` we make use of its `openHandler` to retrieve the needed object lazily (which I think is quite nice)